analytic_engine/src/sst/parquet/meta_data.rs

```diff
@@ -38,68 +45,125 @@ pub enum Error {
 
 define_result!(Error);
 
+const DEFAULT_BLOOM_FILTER_VERSION: u32 = 0;
+
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct RowGroupBloomFilter {
+    // The column filter can be None if the column is not indexed.
+    column_filters: Vec<Option<Bloom>>,
+}
+
+impl RowGroupBloomFilter {
+    pub fn with_num_columns(num_columns: usize) -> Self {
+        Self {
+            column_filters: vec![None; num_columns],
+        }
+    }
+
+    pub fn push_column_filter(&mut self, column_filter: Option<Bloom>) {
+        self.column_filters.push(column_filter);
+    }
+
+    /// Accrue the data belonging to one column.
+    ///
+    /// Caller should ensure the `column_idx` is in the range.
+    pub fn accrue_column_data(&mut self, column_idx: usize, data: &[u8]) {
```

Comment:
How about impl this via IndexMut trait?

Reply:
If so, I guess we have to expose the underlying `ethbloom::Bloom` outside, which seems a little ugly.

Reply:
Didn't you already leak it in push_column_filter above ?

Reply:
Oops. I forget to remove that method.